### ISSUE TITLE
[Suggestion] - Add Falling Blocks game

### ISSUE THREAD

**Issue by zombiepigdragon** [Monika-After-Story/MonikaModDev#3962]
There is a well known game in which players control falling blocks and clear lines made of the tiles of said blocks. While the original can not be included in name due to copyright restrictions, many similar variants could be possible without the legal restrictions: for example, a variation with pentominoes versus the tetrominoes from the well known classic. There are rule variations that add competitive features: for example, clearing lines can affect the opponent's board, or special blocks that give either advantages to those who clear with them or disadvantages to their opponent. I'm willing to implement such a game primarily by myself if given permission, but help would not be turned down. Is this idea feasible?

**Comment by Rai99** [comment]
#647, #2046

Tetris is planned, but probably won't happen for a while on its own. 

Sounds like a pretty big job, but I guess you're free to give it a try. 

Just keep in mind that it should be multiplayer for Monika, fair (ie. feasible for your average person to win at) and it would probably be great if winning wasn't extremely easy. 

**Oh also an important thing I think; It probably shouldn't be too resource demanding** so I'm not sure how far you could take it with those extra blocks and features. 

~~Can we please add puyo, shigu~~

Just let us know if you do plan to start on this and maybe someone like TPG can run through what he's looking for in a Tetris game.

**Comment by AsuraShun** [comment]
With her trolling us every 10 lines pieces

**Comment by zombiepigdragon** [comment]
I want it to be known I have now got a fairly functional completion of Tetris without MAS that uses Pygame and a fairly complete renderer in MAS. If anyone has _any_ suggestions, this is the ideal time to start submitting them.

**Comment by zombiepigdragon** [comment]
I need some better art for this, and if anyone has Easter Egg ideas, that would be great too.

**Comment by Rai99** [comment]
I can't think of how easter eggs could go into Tetris.

**Comment by zombiepigdragon** [comment]
@Rai99 They went into Pong and Hangman just fine

**Comment by Rai99** [comment]
If you named yourself something specific, but Tetris? I can't think of anything.

**Comment by Rai99** [comment]
Anyway, how does it look so far? Is it just basic Tetris or is there anything unique it at all?

Next piece preview up to x blocks?
Holding space?
Back to back?
T-spins?

**Comment by zombiepigdragon** [comment]
@Rai99 Basic for now, but I can easily add a preview (and maybe share it between the player and Monika, for additional annoyance), holding space is even more trivial, and the others are just setting flags if I'm not mistaken.

**Comment by Rai99** [comment]
I was thinking just the basic 1 player Tetris at first, but does Monika play any part in it at all and if so, what does she have to do with it?

**Comment by zombiepigdragon** [comment]
@Rai99 Basically, at the moment, I'm thinking it'll be a simple game of survival. You each play a separate game of Tetris, side by side. I still am not sure whether or not there will be actual effects on each other's boards.

**Comment by AsuraShun** [comment]
What if the easter eggs involve putting the blocks in 3 specific formations where the girls show up when done

**Comment by Rai99** [comment]
Easter eggs, I think should be handled last, after the game is completely finished and practically ready for release. 

@AsuraShun I'm not exactly sure what you mean by formation of blocks. 

For now, my biggest concern is the AI that the player plays against. How do you intend to make it challenging, but fair?

**Comment by AsuraShun** [comment]
I meant a specific pattern of blocks placed on top of each other, doesn't matter which position they go in

**Comment by AsuraShun** [comment]
Like 2 cubes sandwiching an L

**Comment by Rai99** [comment]
I'm not sure what that has to do with any of the girls and I think that has a pretty high chance of happening by accident.

**Comment by zombiepigdragon** [comment]
I'm going to refactor the individual Tetris game, and then put it in its own GitHub repo, so that any individual issues with it can be fixed before it's too integrated with MAS. I'll link that repo when it's finished.

**Comment by ThePotatoGuy** [comment]
[marking as duplicate, but seeing that is in progress, leaving open]

**Comment by zombiepigdragon** [comment]
There's a little bit left, but here's the basics of it: https://github.com/zombiepigdragon/Tetris-VS-AI
Once the first TODO is complete, this will be merged into MAS.

**Comment by zombiepigdragon** [comment]
To clarify: once the AI is beatable, I'm gonna add the code for Tetris into MAS so that Monika is actually a playable opponent.
